.spectral.yml

```diff
@@ -174,6 +175,128 @@ rules:
       functionOptions:
         max: 8
 
+  # SHOULD limit number of sub-resource levels [147]
+  # => https://opensource.zalando.com/restful-api-guidelines/#147
+
+  should-limit-number-of-sub-resource-levels:
+    message: Sub-resource levels should by <= 3
+    description: SHOULD limit number of sub-resource levels [147]
+    documentationUrl: https://opensource.zalando.com/restful-api-guidelines/#147
+    severity: warn
+    given: $.paths.*~
+    then:
+      function: pattern
+      functionOptions:
+        match: ^\/[^\/]*((\/{[^}]*})*\/[^\/]*(\/{[^}]*})*){0,3}\/?$
+
+  # MUST use standard HTTP status codes [150]
+  # => https://opensource.zalando.com/restful-api-guidelines/#150
+
+  must-use-standard-http-status-codes:
+    message: '{{property}} is not a standardized response code'
+    description: MUST use standard HTTP status codes [150]
+    documentationUrl: https://opensource.zalando.com/restful-api-guidelines/#150
+    severity: error
+    given: $.paths.*.*.responses.*~
+    then:
+      function: enumeration
+      functionOptions:
+        values:
+          - '100'
+          - '101'
+          - '200'
+          - '201'
+          - '202'
+          - '203'
+          - '204'
+          - '205'
+          - '206'
+          - '207'
+          - '300'
+          - '301'
+          - '302'
+          - '303'
+          - '304'
+          - '305'
+          - '307'
+          - '400'
+          - '401'
+          - '402'
+          - '403'
+          - '404'
+          - '405'
+          - '406'
+          - '407'
+          - '408'
+          - '409'
+          - '410'
+          - '411'
+          - '412'
+          - '413'
+          - '414'
+          - '415'
+          - '416'
+          - '417'
+          - '423'
+          - '426'
+          - '428'
+          - '429'
+          - '431'
+          - '500'
+          - '501'
+          - '502'
+          - '503'
+          - '504'
+          - '505'
+          - '511'
+          - default
```

Comment:
Should we allow default? I think each response should tell the reader the status code right away, else I have to do future investigation on what's default.

Reply:
Usage of `default` for errors is even encouraged by https://opensource.zalando.com/restful-api-guidelines/#151
> In most cases it is not useful to document all technical errors, especially if they are not under control of the service provider. Thus unless a response code conveys application-specific functional semantics or is used in a none standard way that requires additional explanation, multiple error response specifications can be combined using the following pattern

With rule 151 we also enforce that the `default` response is a problem json.

Reply:
Okey then let's go with default